crates/ruff_cli/src/diagnostics.rs

```diff
@@ -83,7 +91,25 @@ pub fn lint_path(
     };
 
     // Read the file from disk.
-    let contents = std::fs::read_to_string(path)?;
+    let (contents, jupyter_index) = if is_jupyter_notebook(&path) {
+        // See also the black implementation
+        // <https://github.com/psf/black/blob/69ca0a4c7a365c5f5eea519a90980bab72cab764/src/black/__init__.py#L1017-L1046>
+        let notebook: JupyterNotebook = if let Some(notebook) = read_jupyter_notebook(path)? {
+            notebook
+        } else {
+            // Not a python notebook
+            return Ok(Diagnostics {
```

Comment:
Should this be returning some kind of error diagnostic?

Reply:
updated the comment, this could e.g. be an R notebook which we want to just skip